File: cpp/1.two_sum.cpp
#include <unordered_map>
#include <vector>
class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target)
    {
        std::unordered_map<int, int> map;
        
        for (int i = 0; i < nums.size(); i++)
        {
            if (map.find(target - nums[i]) != map.end())
                return {i, map[target - nums[i]]};
            
            map[nums[i]] = i;
        }
        
        return {-1, -1};
    }
};

File: py/1.two_sum.py
import enum
from typing import List
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i, value in enumerate(nums):
            if target - value in map:
                return map[target - value], i
            else:
                map[value] = i
        return -1, -1
